ext/ftp/ftp.c

```diff
@@ -707,6 +707,61 @@ ftp_mlsd(ftpbuf_t *ftp, const char *path, const size_t path_len)
 }
 /* }}} */
 
+/* {{{ ftp_mlsd_parse_line
+ */
+int
+ftp_mlsd_parse_line(zval entry, const char *line) {
+
+	int linelength = strlen(line);
+	int chunklength = 0;
+	int mode = FTP_MLSD_MODE_NAME;
+
+	char name[FTP_MLSD_CHUNK_LENGTH_MAX + 1] = "";
+
+	zval zstr;
+
+	if(linelength > FTP_MLSD_LINE_LENGTH_MAX) {
+		php_error_docref(NULL, E_WARNING, "Line of entry '%s' exceed the limit of %u bytes. Skip entry.", line, FTP_MLSD_LINE_LENGTH_MAX);
+		return 1;
+	}
+
+	// Copy the line to input to not modify the original.
+	char *input = malloc(linelength);
+	strncpy(input, line, linelength);
+
+	while(strlen(input)) {
```

Comment:
strlen() loops over the entire string to find the end. If you want to check whether a string is non-empty, you can use `while (*input)` or `while (*input != '\0')` to be more explicit.